### PR TITLE
One view query instead of one per global ID

### PR DESCRIPTION
**Pull request by millerdev**
There are 16 global fixture data types for UATBC, and the fixture code had been executing a separate couch query for each one. My rough time measurements showed that that can take up to 20s. Combining them all in a single query takes ~2.5s. The 20s time could be offset somewhat by caching, but I think this is still a win because it should make the slow (empty cache) case faster. Also, the results are now all cached together so it may even improve cache effects.

The combined query currently returns nearly 1000 items, so it still looks promising to not send item set fixtures that have not changed since the user's last sync. Just compiling the XML for that many items can take a while.

@proteusvacuum @dannyroberts

### CODE REVIEW

File: corehq/apps/fixtures/models.py [hunk 1]
```diff
@@ -409,8 +409,12 @@ def by_group(cls, group, wrap=True):
 
     @classmethod
     def by_data_type(cls, domain, data_type, bypass_cache=False):
-        data_type_id = _id_from_doc(data_type)
-        return get_fixture_items_by_data_type(domain, data_type_id, bypass_cache)
+        return cls.by_data_types(domain, [data_type], bypass_cache)
+
+    @classmethod
+    def by_data_types(cls, domain, data_types, bypass_cache=False):
+        data_type_ids = frozenset(_id_from_doc(d) for d in data_types)
```
Comment (dannyroberts) [inline corehq/apps/fixtures/models.py]: Why'd you have to use `frozenset` here, rather than `set`? Also seems related to build failure

**Comment by dannyroberts** [comment]
```
ERROR: corehq.apps.commtrack.tests.test_xml:CommTrackOTATest.test_force_consumption
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 267, in inner
    return helper(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 238, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 253, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 234, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/commtrack/tests/test_xml.py", line 186, in test_force_consumption
    balance_blocks = _get_ota_balance_blocks(self.domain, self.user)
  File "/mnt/commcare-hq-ro/corehq/apps/commtrack/tests/test_xml.py", line 725, in _get_ota_balance_blocks
    return util.extract_balance_xml(restore_config.get_payload().as_string())
  File "corehq/ex-submodules/casexml/apps/phone/restore.py", line 703, in get_payload
    response = self.generate_payload()
  File "corehq/ex-submodules/casexml/apps/phone/restore.py", line 709, in generate_payload
    response = self._generate_restore_response(async_task=async_task)
  File "corehq/ex-submodules/casexml/apps/phone/restore.py", line 758, in _generate_restore_response
    for element in provider.get_elements(self.restore_state):
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/standard.py", line 61, in get_elements
    elements = provider(restore_state)
  File "/mnt/commcare-hq-ro/corehq/apps/fixtures/fixturegenerators.py", line 57, in __call__
    items = FixtureDataItem.by_data_types(restore_user.domain, global_types)
  File "/mnt/commcare-hq-ro/corehq/apps/fixtures/models.py", line 417, in by_data_types
    return get_fixture_items_for_data_types(domain, data_type_ids, bypass_cache)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache.py", line 40, in inner
    return helper(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache_helper.py", line 145, in __call__
    return self.call(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache_helper.py", line 55, in call
    key = self.get_cache_key(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache_helper.py", line 126, in get_cache_key
    for value in values)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache_helper.py", line 126, in <genexpr>
    for value in values)
  File "/vendor/lib/python2.7/site-packages/quickcache/quickcache_helper.py", line 112, in _serialize_for_key
    raise ValueError('Bad type "{}": {}'.format(type(value), value))
ValueError: Bad type "<type 'frozenset'>": frozenset([])
```

**Comment by dannyroberts** [comment]
Oh and also—great optimization!

File: corehq/apps/fixtures/models.py (hunk 1 above)
Comment (millerdev) [inline corehq/apps/fixtures/models.py]: huh, I thought frozenset would be pickle-able, which I thought would be necessary for quickcache. I'll take a look.

Comment (dannyroberts) [inline corehq/apps/fixtures/models.py]: Ah, makes sense. I think `@memoized` _does_ require that, but quickcache is intentionally made to do its own hashing based on something closer to what we'd intuitively consider the "value" of an object. So `set(1, 2, 3)` and `set(2, 3, 1)`—separately instantiated—have the same hash. See https://github.com/dimagi/quickcache/blob/4e608301688daef7c0d026230ae7ca4a95d7b957/quickcache/quickcache_helper.py#L106 for the code if you're interested.

Comment (dannyroberts) [inline corehq/apps/fixtures/models.py]: We could add support for frozen set to that I suppose—I think it's just never come up before

**Comment by millerdev** [comment]
✅

**Comment by proteusvacuum** [comment]
🎉

**Pull request merged**